plugins/org.jkiss.dbeaver.ui.ai/src/org/jkiss/dbeaver/ui/ai/engine/copilot/CopilotConfigurator.java

```diff
@@ -83,14 +86,33 @@ public void createControl(
     @Override
     public void loadSettings(@NotNull LegacyAISettings<CopilotProperties> configuration) {
         token = CommonUtils.toString(configuration.getProperties().getToken());
-        model = readModel(configuration).getName();
+        model = CommonUtils.toString(configuration.getProperties().getModel());
         temperature = CommonUtils.toString(configuration.getProperties().getTemperature(), "0.0");
         logQuery = CommonUtils.toBoolean(configuration.getProperties().isLoggingEnabled());
         accessToken = CommonUtils.toString(configuration.getProperties().getToken(), "");
         accessTokenText.setText(accessToken);
+        getModels(false);
         applySettings();
     }
 
+    private void getModels(boolean forceRefresh) {
```

Comment:
Confusing method name. Something like `populateModelsCombo` would suit nicer, don't you think?